deployer panic

```text
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4b7a5b]

goroutine 25 [running]:
github.com/flynn/flynn/controller/deployer/strategies.waitForJobEvents(0xc2080fa8a0, 0xc2080fa840, 0xc208039740, 0x0, 0x0)
	/home/ubuntu/go/src/github.com/flynn/flynn/controller/deployer/strategies/main.go:57 +0xc8b
github.com/flynn/flynn/controller/deployer/strategies.allAtOnce(0x7f0f846deef0, 0xc20801c0c0, 0xc208036038, 0xc20802b0e0, 0xc2080fa840, 0x0, 0x0)
	/home/ubuntu/go/src/github.com/flynn/flynn/controller/deployer/strategies/all-at-once.go:46 +0xeda
main.(*context).HandleJob(0xc20801cba0, 0xc2080a4280, 0x0, 0x0)
	/home/ubuntu/go/src/github.com/flynn/flynn/controller/deployer/main.go:120 +0x1845
main.*context.HandleJobÂ·fm(0xc2080a4280, 0x0, 0x0)
	/home/ubuntu/go/src/github.com/flynn/flynn/controller/deployer/main.go:59 +0x43
github.com/flynn/flynn/Godeps/_workspace/src/github.com/bgentry/que-go.(*Worker).WorkOne(0xc2080326c0, 0xc20802a901)
	/home/ubuntu/go/src/github.com/flynn/flynn/Godeps/_workspace/src/github.com/bgentry/que-go/worker.go:103 +0x481
github.com/flynn/flynn/Godeps/_workspace/src/github.com/bgentry/que-go.(*Worker).Work(0xc2080326c0)
	/home/ubuntu/go/src/github.com/flynn/flynn/Godeps/_workspace/src/github.com/bgentry/que-go/worker.go:74 +0x1b1
created by github.com/flynn/flynn/Godeps/_workspace/src/github.com/bgentry/que-go.(*WorkerPool).Start
	/home/ubuntu/go/src/github.com/flynn/flynn/Godeps/_workspace/src/github.com/bgentry/que-go/worker.go:164 +0x1d6
```

https://s3.amazonaws.com/flynn-ci-logs/20150128015158-b67d3a73-build-5b73a373e92015e9581b9032fe0bb94665aae803-2015-01-28-02-19-31.txt

Looks like the channel was closed (check the comma ok).

--- a/controller/deployer/strategies/main.go
+++ b/controller/deployer/strategies/main.go
@@ -50,10 +50,18 @@ type jobEvents map[string]map[string]map[string]int
 func waitForJobEvents(events chan *ct.JobEvent, deployEvents chan<- ct.DeploymentEvent, expected jobEvents) error {
 	fmt.Printf("waiting for job events: %v\n", expected)
 	actual := make(jobEvents)
+outer:
 	for {
 	inner:
 		select {
-		case event := <-events:
+		case event, ok := <-events:
+			if !ok {
+				// if this happens, it means defer cleanup is in progress
+
+				// TODO: this could also happen if the stream connection
+				// dropped. handle that case
+				break outer
+			}
 			fmt.Printf("got job event: %s %s %s\n", event.Type, event.JobID, event.State)
 			if _, ok := actual[event.Job.ReleaseID]; !ok {
 				actual[event.Job.ReleaseID] = make(map[string]map[string]int)
@@ -94,4 +102,5 @@ func waitForJobEvents(events chan *ct.JobEvent, deployEvents chan<- ct.Deploymen
 			return fmt.Errorf("timed out waiting for job events: ", expected)
 		}
 	}
+	return nil
 }

